### ISSUE TITLE
Directline sdk is broken without installing certain packages. Please address

### ISSUE THREAD

**Issue by xtianus79** [microsoft/BotFramework-DirectLineJS#288]
Here is the errors from rxjs not working

```javascript
ERROR in ./node_modules/botframework-directlinejs/lib/directLineStreaming.js
Module not found: Error: Can't resolve 'rxjs/BehaviorSubject' in '/mnt/c/repos/ai-xxxxx-frontend-website/node_modules/botframework-directlinejs/lib'

ERROR in ./node_modules/botframework-directlinejs/lib/directLine.js
Module not found: Error: Can't resolve 'rxjs/BehaviorSubject' in '/mnt/c/repos/ai-xxxxx-frontend-website/node_modules/botframework-directlinejs/lib'

ERROR in ./node_modules/botframework-directlinejs/lib/directLineStreaming.js
Module not found: Error: Can't resolve 'rxjs/Observable' in '/mnt/c/repos/ai-xxxxxx-frontend-website/node_modules/botframework-directlinejs/lib'

ERROR in ./node_modules/botframework-directlinejs/lib/directLine.js
Module not found: Error: Can't resolve 'rxjs/Observable' in '/mnt/c/repos/ai-xxxxx-frontend-website/node_modules/botframework-directlinejs/lib'

ERROR in ./node_modules/botframework-directlinejs/lib/directLine.js
Module not found: Error: Can't resolve 'rxjs/add/observable/dom/ajax' in '/mnt/c/repos/ai-xxxxxx-frontend-website/node_modules/botframework-directlinejs/lib'

ERROR in ./node_modules/botframework-directlinejs/lib/directLine.js
Module not found: Error: Can't resolve 'rxjs/add/observable/empty' in '/mnt/c/repos/ai-xxxxx-frontend-website/node_modules/botframework-directlinejs/lib'

ERROR in ./node_modules/botframework-directlinejs/lib/directLine.js
Module not found: Error: Can't resolve 'rxjs/add/observable/from' in '/mnt/c/repos/ai-xxxxx-frontend-website/node_modules/botframework-directlinejs/lib'

ERROR in ./node_modules/botframework-directlinejs/lib/directLine.js
Module not found: Error: Can't resolve 'rxjs/add/observable/interval' in '/mnt/c/repos/ai-xxxxxxx-frontend-website/node_modules/botframework-directlinejs/lib'

ERROR in ./node_modules/botframework-directlinejs/lib/directLine.js
Module not found: Error: Can't resolve 'rxjs/add/observable/of' in '/mnt/c/repos/ai-xxxxx-frontend-website/node_modules/botframework-directlinejs/lib'

ERROR in ./node_modules/botframework-directlinejs/lib/directLine.js
Module not found: Error: Can't resolve 'rxjs/add/observable/throw' in '/mnt/c/repos/ai-xxxxxxx-frontend-website/node_modules/botframework-directlinejs/lib'

ERROR in ./node_modules/botframework-directlinejs/lib/directLine.js
Module not found: Error: Can't resolve 'rxjs/add/operator/catch' in '/mnt/c/repos/ai-xxxxxx-frontend-website/node_modules/botframework-directlinejs/lib'

ERROR in ./node_modules/botframework-directlinejs/lib/directLine.js
Module not found: Error: Can't resolve 'rxjs/add/operator/combineLatest' in '/mnt/c/repos/ai-xxxxxx-frontend-website/node_modules/botframework-directlinejs/lib'

ERROR in ./node_modules/botframework-directlinejs/lib/directLine.js
Module not found: Error: Can't resolve 'rxjs/add/operator/concatMap' in '/mnt/c/repos/ai-xxxxxx-frontend-website/node_modules/botframework-directlinejs/lib'

ERROR in ./node_modules/botframework-directlinejs/lib/directLine.js
Module not found: Error: Can't resolve 'rxjs/add/operator/count' in '/mnt/c/repos/ai-xxxxxx-frontend-website/node_modules/botframework-directlinejs/lib'

ERROR in ./node_modules/botframework-directlinejs/lib/directLine.js
Module not found: Error: Can't resolve 'rxjs/add/operator/delay' in '/mnt/c/repos/ai-xxxxxxxx-frontend-website/node_modules/botframework-directlinejs/lib'

ERROR in ./node_modules/botframework-directlinejs/lib/directLine.js
Module not found: Error: Can't resolve 'rxjs/add/operator/do' in '/mnt/c/repos/ai-xxxxx-frontend-website/node_modules/botframework-directlinejs/lib'

ERROR in ./node_modules/botframework-directlinejs/lib/directLine.js
Module not found: Error: Can't resolve 'rxjs/add/operator/filter' in '/mnt/c/repos/ai-xxxxx-frontend-website/node_modules/botframework-directlinejs/lib'

ERROR in ./node_modules/botframework-directlinejs/lib/directLine.js
Module not found: Error: Can't resolve 'rxjs/add/operator/map' in '/mnt/c/repos/ai-xxxxx-frontend-website/node_modules/botframework-directlinejs/lib'

ERROR in ./node_modules/botframework-directlinejs/lib/directLine.js
Module not found: Error: Can't resolve 'rxjs/add/operator/mergeMap' in '/mnt/c/repos/ai-xxxxxxx-frontend-website/node_modules/botframework-directlinejs/lib'

ERROR in ./node_modules/botframework-directlinejs/lib/directLine.js
Module not found: Error: Can't resolve 'rxjs/add/operator/retryWhen' in '/mnt/c/repos/ai-xxxxxxxx-frontend-website/node_modules/botframework-directlinejs/lib'

ERROR in ./node_modules/botframework-directlinejs/lib/directLine.js
Module not found: Error: Can't resolve 'rxjs/add/operator/share' in '/mnt/c/repos/ai-xxxxxx-frontend-website/node_modules/botframework-directlinejs/lib'

ERROR in ./node_modules/botframework-directlinejs/lib/directLine.js
Module not found: Error: Can't resolve 'rxjs/add/operator/take' in '/mnt/c/repos/ai-xxxxxx-frontend-website/node_modules/botframework-directlinejs/lib'

ERROR in ./node_modules/botframework-directlinejs/lib/directLine.js
Module not found: Error: Can't resolve 'rxjs/scheduler/async' in '/mnt/c/repos/ai-xxxxxxxx-frontend-website/node_modules/botframework-directlinejs/lib'
```

After installing rxjs-compat or I'm assuming rxjs 5.0 
these are the remaining errors
```javascript
ERROR in ./node_modules/botframework-streaming/browser/webSocket/nodeWebSocket.js
Module not found: Error: Can't resolve 'crypto' in '/mnt/c/repos/ai-xxxxxx-frontend-website/node_modules/botframework-streaming/browser/webSocket'
ERROR in ./node_modules/botframework-streaming/browser/webSocket/nodeWebSocket.js
Module not found: Error: Can't resolve 'http' in '/mnt/c/repos/ai-xxxxxxxxx-frontend-website/node_modules/botframework-streaming/browser/webSocket'
ERROR in ./node_modules/botframework-streaming/browser/subscribableStream.js
Module not found: Error: Can't resolve 'stream' in '/mnt/c/repos/ai-xxxxxxxx-frontend-website/node_modules/botframework-streaming/browser'
```

**Comment by compulim** [comment]
@xtianus79 what bundler are you using? I am guessing it is not Webpack, so it don't turn `crypto`/`http`/`stream` into `node-libs-browser`.

@stevengum I think we need to take away Node.js dependencies from `botframework-streaming`.

**Comment by xtianus79** [comment]
@compulim I am using Angular cli 9+ and to make matters super interesting Angular universal. lol I always have to decide if that is a culprit.

**Comment by compulim** [comment]
@xtianus79 I think there are two issues:

We assume Webpack is everything. But in fact, Webpack is too magical (automatically convert Node.js package into [`node-libs-browser`](https://npmjs.com/package/node-libs-browser)). @stevengum and @Zerryth are trying to take away Node.js package or directly linking to [`node-libs-browser`](https://npmjs.com/package/node-libs-browser)

Another issue is the version of RxJS we are using is too old (`5.5.10`). May be it conflict with the version Angular brought in.

We are not very familiar with Angular. Could you provide a minimal repos that can build and host Web Chat on Angular? So we could check that into our repos and make sure we our integration test will test basic functionality on Web Chat on Angular?

**Comment by stevengum** [comment]
Closed #289 as the same error message was being reported here. Additionally, I have the same concerns about the missing `node-libs-browser`.

> _@stevengum I am using in the project 3.8.3 but the streaming 3.5.2 is causing the issue I believe._

_@xtianus79 regarding version of TypeScript used ([comment](https://github.com/microsoft/BotFramework-DirectLineJS/issues/289#issuecomment-648946630))_

**Comment by stevengum** [comment]
Quick update from my end for botframework-streaming:

I’m currently working on removing the node.js specific dependencies and either replacing it with an explicit pollyfill or creating a Node/Browser specific implementation classes.

I have a simple manual integration test that I’m using to validate the work I’m doing.

This work will ship with R10.

**Comment by stevengum** [comment]
Unfortunately there isn't enough R10 bake time for the removal of the Node.js dependencies in the botframework-streaming.

The currently targeted milestone is R11.

**Comment by stevengum** [comment]
The `crypto`, `http`, and `stream` dependency resolution issue has been fixed via https://github.com/microsoft/BotFramework-DirectLineJS/pull/323.

**Comment by stevengum** [comment]
The fix for resolving Node.js dependencies was just released in version [0.14.1](https://www.npmjs.com/package/botframework-directlinejs/v/0.14.1) of `botframework-directlinejs`.

**Comment by corinagum** [comment]
Closing as resolved.
